audioclips/src/main/java/org/odk/collect/audioclips/AudioClipViewModel.kt

```diff
@@ -85,6 +86,7 @@ class AudioClipViewModel(private val mediaPlayerFactory: Supplier<MediaPlayer>,
     private fun playNext(playlist: Queue<Clip>) {
         val nextClip = playlist.poll()
         if (nextClip != null) {
+            pause()
```

Comment:
Looking at the state diagram at https://developer.android.com/reference/android/media/MediaPlayer, I think really we should be calling `MediaPlayer#stop` in `AudioClipViewModel#cleanUpAfterClip`. I think when I initially built this I assumed that `onCompletion` being called would mean that we didn't need a `stop()`, but it seems like the pause helping here would suggest that and that lines up with the "Stopped" state in the docs' charts.

We'd definitely want a test for this in `AudioClipViewModelTest`. It looks like the approach there is to use `InOrder` verifiers to check that the `MediaPlayer` is moved through he states correctly.

Reply:
> Looking at the state diagram at https://developer.android.com/reference/android/media/MediaPlayer, I think really we should be calling MediaPlayer#stop in AudioClipViewModel#cleanUpAfterClip. I think when I initially built this I assumed that onCompletion being called would mean that we didn't need a stop(), but it seems like the pause helping here would suggest that and that lines up with the "Stopped" state in the docs' charts.

I was thinking about that. The problem is that stops moves the seekbar to the starting position and that's nat what we had in v2023.3.1.

Reply:
> The problem is that stops moves the seekbar to the starting position and that's nat what we had in v2023.3.1.

That feels correct though I think? I guess you could argue either way whether the seeker should move to the beginning or the end when payback stops. We could also make it so the seekbar is always set to the end if the play is "stopped" rather than "paused" regardless of position.

Reply:
I'm not a huge fan of the way it worked in v2023.3 but didn't want to introduce changes. If you think it's ok I can agree I guess. I will play with it.

Reply:
Done.